analysis/model/src/main/kotlin/de/maibornwolff/codecharta/serialization/NodeJsonDeserializer.kt

```diff
@@ -45,16 +45,16 @@ internal class NodeJsonDeserializer : JsonDeserializer<Node> {
         return nameElement.asString
     }
 
-    private fun deserializeAttributes(context: JsonDeserializationContext, jsonNode: JsonObject): Map<String, Any> {
+    private fun deserializeAttributes(jsonNode: JsonObject): Map<String, Any> {
         val attributes = jsonNode.get("attributes") ?: return mapOf()
 
         val gson = GsonBuilder().create()
         return gson.fromJson<Map<String, Any>>(attributes, Map::class.java)
     }
 
-    private fun deserializeChildren(context: JsonDeserializationContext, jsonNode: JsonObject): List<Node> {
+    private fun deserializeChildren(jsonNode: JsonObject): List<Node> {
         val children = jsonNode.get("children") ?: return listOf()
 
-        return children.asJsonArray.mapTo(ArrayList()) { deserialize(it, Node::class.java, context) }
+        return children.asJsonArray.mapTo(ArrayList()) { deserialize(it, Node::class.java, null) }
```

Comment:
What was the purpose of the context? Why is it never required?

Reply:
It was never used before. I am not certain what it was meant to be used for. It is just required by the interface, that's why I kept the argument.